### PR TITLE
chore(core): add groups to HTTP authenticator and tweak HTTP client

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Adds non-chunked response support to `HttpClient`.
Also tweaks the client to make it reusable for multiple request execution.

-------------

This PR adds a new `/settings` endpoint.
The endpoint can expose part of the config in JSON format.
The web console will use this endpoint, it is a way of sharing configuration.

The PR also changes:
- the behaviour of StaticProcessor, which will require authentication only if the auth type is Basic Authentication
- the HTTP response in case of authentication failures, the `WWW-Authenticate: Basic realm="questdb"` header is sent only if the auth type is Basic Authentication

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 63 / 100 (63.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|31|41|75.61%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 108 (61.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|34|49|69.39%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 66 / 108 (61.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|34|49|69.39%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 140 / 191 (73.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|34|49|69.39%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/SettingsProcessor.java|16|21|76.19%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|14|16|87.50%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpPingProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextImportProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TableStatusCheckProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 146 / 199 (73.37%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|37|54|68.52%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/SettingsProcessor.java|16|21|76.19%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|14|16|87.50%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpPingProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextImportProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TableStatusCheckProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/http/client/HttpClient.java [hunk 1]
```diff
@@ -75,35 +78,48 @@ public void close() {
...
+            this.socket = socketFactory.newInstance(nf, LOG);
+            this.host = host;
+            this.port = port;
+        } else {
+            // Connection reuse.
+            if (!Chars.equals(this.host, host)) {
+                throw new HttpClientException("same host expected for connection reuse [host=").put(this.host).put(']');
+            }
+            if (this.port != port) {
+                throw new HttpClientException("same host expected for connection reuse [port=").put(this.port).put(']');
```
Comment (glasstiger) [inline core/src/main/java/io/questdb/cutlass/http/client/HttpClient.java:99]:
```suggestion
                throw new HttpClientException("same port expected for connection reuse [port=").put(this.port).put(']');
```

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 146 / 199 (73.37%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|37|54|68.52%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/SettingsProcessor.java|16|21|76.19%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|14|16|87.50%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpPingProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextImportProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TableStatusCheckProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

File: core/src/main/java/io/questdb/cutlass/http/client/HttpClient.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/cutlass/http/client/HttpClient.java]: Thanks for spotting the typo. Fixed in 5f1b452ef9cafe595241a4015d7596c97746600d

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 146 / 199 (73.37%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|37|54|68.52%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/SettingsProcessor.java|16|21|76.19%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|14|16|87.50%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpPingProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextImportProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TableStatusCheckProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 142 / 183 (77.60%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/SettingsProcessor.java|16|21|76.19%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|34|39|87.18%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpPingProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/http/LineHttpSender.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextImportProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TableStatusCheckProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 143 / 184 (77.72%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClientMain.java|0|20|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/AnonymousHttpAuthenticator.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/SettingsProcessor.java|16|21|76.19%|
|:large_blue_circle:|io/questdb/cutlass/http/client/AbstractResponse.java|23|29|79.31%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpConnectionContext.java|10|12|83.33%|
|:large_blue_circle:|io/questdb/cutlass/http/client/HttpClient.java|34|39|87.18%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpHeaderParser.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpRequestProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/HealthCheckProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/LineHttpPingProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/HttpServer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/http/LineHttpSender.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextImportProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/StaticContentProcessor.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TableStatusCheckProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/PrometheusMetricsProcessor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/network/JavaTlsClientSocket.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|2|2|100.00%|

**Review by glasstiger — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@glasstiger thanks for the review!

**Pull request merged**